### DISCUSSION TITLE
AttributeError: 'DenseBlock' object has no attribute 'get_adapter'

### DISCUSSION THREAD

**Discussion by yurimarimo** [keras-team/autokeras#1487] [Q&A] [answered]
This is my code.

```
import autokeras as ak

input_node = ak.Input()
gru_node = ak.RNNBlock(return_sequences=True, layer_type="gru")(input_node)
output_node = ak.DenseBlock()(gru_node)

auto_model = ak.AutoModel(
    inputs=input_node, 
    outputs=output_node,
    overwrite=True,
    max_trials=2)

auto_model.fit(X_train, y_train, batch_size=100, epochs=1, validation_split=0.1)
```

When I tried, I got the following error.
```
AttributeError: 'DenseBlock' object has no attribute 'get_adapter'
```

I don't know how to solve this problem, so I hope you can tell me something.

Thank you.

**Comment by KiLJ4EdeN** [comment, marked as answer]
> This is my code.
> 
> ```
> import autokeras as ak
> 
> input_node = ak.Input()
> gru_node = ak.RNNBlock(return_sequences=True, layer_type="gru")(input_node)
> output_node = ak.DenseBlock()(gru_node)
> 
> auto_model = ak.AutoModel(
>     inputs=input_node, 
>     outputs=output_node,
>     overwrite=True,
>     max_trials=2)
> 
> auto_model.fit(X_train, y_train, batch_size=100, epochs=1, validation_split=0.1)
> ```
> 
> When I tried, I got the following error.
> 
> ```
> AttributeError: 'DenseBlock' object has no attribute 'get_adapter'
> ```
> 
> I don't know how to solve this problem, so I hope you can tell me something.
> 
> Thank you.

Hello I reproduced your error with random vectors,

The error can be solved by adding a head to the end of the network, this also can be a RegressionHead.
A full example would look like:

```python
import autokeras as ak

input_node = ak.Input()
gru_node = ak.RNNBlock(return_sequences=True, layer_type="lstm")(input_node)
dense_node = ak.DenseBlock()(gru_node)
output_node = ak.ClassificationHead()(dense_node)

auto_model = ak.AutoModel(
inputs=input_node,
outputs=output_node,
overwrite=True,
max_trials=2)

auto_model.fit(X_train, y_train, batch_size=100, epochs=1, validation_split=0.1)
```

I think the problem is that you are using a non final layer as an output layer.

**Comment by yurimarimo** [comment]
Thank you for your answer.
I've solved the problem.
